query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/RequestUtils.java

```diff
@@ -164,29 +189,85 @@ public ReportImpl endpointIsReachable(UrlField urlField) {
         return report;
     }
 
-    private WebClient generateClient(String url, CredentialsField creds,
-            Map<String, String> queryParams) {
-        SecureCxfClientFactory<WebClient> clientFactory = createFactory(url, creds);
+    public class WebClientBuilder {
+
+        private final WebClient webClient;
 
-        WebClient client = clientFactory.getClient();
+        private WebClientBuilder(String url, String username, String password,
```

Comment:
These parameters don't make a lot of sense to me. why would you want to take in the username password and also the subject? I thought the point of the username and password was to create a subject. It seems like the subject should be removed since everyone is passing null anyways.